### PR TITLE
Add Typescript typing

### PR DESCRIPTION
**Pull request by hoetmaaiers**

### CODE REVIEW

**Comment by hoetmaaiers** [comment]
Please review or approve this PR.

**Comment by aflegel** [comment]
It might be faster/better to move this to Definitely Typed, they have some [contribution guidelines](https://definitelytyped.org/guides/contributing.html).

**Comment by zacbdev** [comment]
@alex3165 Can you please review this PR and merge? I need it definitely. Thank you.

**Comment by mindflay3r** [comment]
Can you guys use more verbose types than `Function`? eslint gets mad when you use it, saying:
```
Don't use `Function` as a type. The `Function` type accepts any function-like value.

It provides no type safety when calling the function, which can be a common source of bugs.

It also accepts things like class declarations, which will throw at runtime as they will not be called with `new`.

If you are expecting the function to accept certain arguments, you should explicitly define the function shape  @typescript-eslint/ban-types
```

**Comment by everestkid** [comment]
Nice to have typscript

File: src/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+import type { DrawOptions, EditOptions, ControlPosition } from 'leaflet';
+
+
+interface EditControlProps {
+  onEdited?: Function;
```
Comment (srghma) [inline src/index.d.ts:6]: agree with @mindflay3r 

could we replace the Function type with something more verbose

tnx

**Review by sinnrrr — APPROVED** [review]

**Comment by hoetmaaiers** [comment]
Thank you, I realise this is just a start...

**Pull request merged**